### PR TITLE
add DatasetBase.yield_records

### PR DESCRIPTION
**Pull request by TomGoBravo**
## This PR

* adds DatasetBase.yield_records, replacing `records` property that breaks in a future PR
* replaces use of `get_us_latest_for_state` with 2 digit code passed to `get_records_for_fips` 
* adds `test_fill_fields_with_data_source_nan_overwrite` that exercises the unexpected current behavior when combining datasets
* make get_subset aggregation_level parameter optional, it was required
* add `_` prefix to some functions that are internal to a class

### Please Check
- [ ] Are the [JSON schemas](https://github.com/covid-projections/covid-data-model/tree/master/api/schemas) updated?
- [ ] Are the [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.V1.md) updated?
- [ ] Are tests passing?

### CODE REVIEW

File: libs/datasets/timeseries.py [hunk 1]
```diff
@@ -74,16 +74,16 @@ def latest_values(self, aggregation_level=None) -> pd.DataFrame:
 
         Return: DataFrame
         """
-        if not aggregation_level:
+        if aggregation_level is None:
```
Comment (ghop02) [inline libs/datasets/timeseries.py:77]: Why this change?

File: libs/datasets/timeseries.py [hunk 2]
```diff
@@ -74,16 +74,16 @@ def latest_values(self, aggregation_level=None) -> pd.DataFrame:
...
             county = self.latest_values(aggregation_level=AggregationLevel.COUNTY)
             state = self.latest_values(aggregation_level=AggregationLevel.STATE)
             return pd.concat([county, state])
-
-        if aggregation_level == AggregationLevel.COUNTY:
+        elif aggregation_level == AggregationLevel.COUNTY:
```
Comment (ghop02) [inline libs/datasets/timeseries.py:81]: apparently the preferred way of comparing enums is with `is`.  i.e. `aggregation_level is AggregationLevel.COUNTY`

File: pyseir/parameters/parameter_ensemble_generator.py [hunk 3]
```diff
@@ -39,10 +39,9 @@ def __init__(self, fips, N_samples, t_list, I_initial=1, suppression_policy=None
                 load_data.load_county_metadata().set_index("fips").loc[fips].to_dict()
             )
             self.state_abbr = us.states.lookup(self.county_metadata["state"]).abbr
-            self._latest = combined_datasets.get_us_latest_for_fips(self.fips)
         else:
             self.state_abbr = us.states.lookup(fips).abbr
```
Comment (ghop02) [inline pyseir/parameters/parameter_ensemble_generator.py:43]: Are `self.state_abbr` and `self.county_metadata` used anywhere else? You may be able to remove this whole block

File: libs/datasets/timeseries.py (hunk 2 above)
Comment (TomGoBravo) [reply libs/datasets/timeseries.py:81]: Done

File: libs/datasets/timeseries.py (hunk 1 above)
Comment (TomGoBravo) [reply libs/datasets/timeseries.py:77]: No particular reason other than touching code to do with aggregation_level and thinking it is a little cleaner, but I can also see why having the `is None` on its own is better.

File: pyseir/parameters/parameter_ensemble_generator.py (hunk 3 above)
Comment (TomGoBravo) [reply pyseir/parameters/parameter_ensemble_generator.py:43]: Good call! done.
